src/vm/dllimportcallback.cpp

```diff
@@ -1412,6 +1412,8 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
+    if (pStubMD == NULL)
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
```

Comment:
Mutating state is recipe for problems.

It would be better to compute the right number of arguments here and set it here in all cases; and move this:

```
    m_cbActualArgSize =
        (pStubMD != NULL) ? pStubMD->AsDynamicMethodDesc()->GetNativeStackArgSize() : pMD->SizeOfArgStack();
```

to the `#else` part of the ifdef.